src/Region/Region.cc

```diff
@@ -388,50 +389,94 @@ casacore::LCRegion* Region::GetImageRegion(
+            double ref_length_ratio;
+            if (region_type == CARTA::RegionType::ELLIPSE) {
+                ref_length_ratio = x_length / y_length;
+            } else {
+                ref_length_ratio = y_length / x_length;
+            }
 
-    // Make 4-point polygon, add center point, then convert them all to output image pixels
-    // points = [p0, p1, p2, p3, center]
-    std::vector<CARTA::Point> points = GetApproximateEllipsePoints(4);
-    points.push_back(center_point);
-    casacore::Vector<casacore::Double> x, y;
+            std::vector<CARTA::Point> points;
+            if (region_type == CARTA::RegionType::ELLIPSE) {
+                // Make polygon with 4 points
+                points = GetApproximateEllipsePoints(4);
+            } else {
+                // Get midpoints of 4 sides of rectangle
+                points = GetRectangleMidpoints();
+            }
 
-    if (ConvertPolygonToImage(points, output_csys, x, y)) {
-        // vector0 is (center, p0), vector1 is (center, p1)
-        auto v0_delta_x = x[0] - x[4];
-        auto v0_delta_y = y[0] - y[4];
-        auto v1_delta_x = x[1] - x[4];
-        auto v1_delta_y = y[1] - y[4];
-
-        auto v0_length = sqrt((v0_delta_x * v0_delta_x) + (v0_delta_y * v0_delta_y));
-        auto v1_length = sqrt((v1_delta_x * v1_delta_x) + (v1_delta_y * v1_delta_y));
-        double converted_length_ratio = v1_length / v0_length;
-
-        // Compare converted to reference length ratio
-        double length_ratio_difference = fabs(ref_length_ratio - converted_length_ratio);
-        spdlog::debug("Ellipse distortion check: length ratio difference={:.3e}", length_ratio_difference);
-        if (length_ratio_difference > 1e-4) {
+            // add center point; points = [p0, p1, p2, p3, center]
+            points.push_back(center_point);
+
+            // convert points to output image pixels
+            casacore::Vector<casacore::Double> x, y;
+            if (ConvertPolygonToImage(points, output_csys, x, y)) {
+                // vector0 is (center, p0), vector1 is (center, p1)
+                auto v0_delta_x = x[0] - x[4];
+                auto v0_delta_y = y[0] - y[4];
+                auto v1_delta_x = x[1] - x[4];
+                auto v1_delta_y = y[1] - y[4];
+
+                auto v0_length = sqrt((v0_delta_x * v0_delta_x) + (v0_delta_y * v0_delta_y));
+                auto v1_length = sqrt((v1_delta_x * v1_delta_x) + (v1_delta_y * v1_delta_y));
+                double converted_length_ratio = v1_length / v0_length;
+
+                // Compare converted to reference length ratio
+                double length_ratio_difference = fabs(ref_length_ratio - converted_length_ratio);
+                spdlog::debug("{} distortion check: length ratio difference={:.3e}", RegionName(region_type), length_ratio_difference);
+                if (length_ratio_difference < 1e-4) {
```

Comment:
Regarding this arbitrary 1e-4 threshold for the length ratio difference, it works fine so far. 👍  It may need quite some case studies to verify our current choice but for now I suggest we just adopt it.